Server/pathfinder/pathfinder.java

```diff
@@ -0,0 +1,219 @@
+package se.uu.it.runestone.teamone.pathfinding;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.HashMap;
+import java.util.ArrayList;
+
+import se.uu.it.runestone.teamone.pathfinding.graph;
+
+/**
+ * A class that implements pathfinding algorithms.
+ * 
+ * @author Åke Lagercrantz
+ */
+public class PathFinder {
+  /// A FrontierNode comparator that compares nodes on priority.
+  private PathFinder.FrontierNodeComparator frontierComparator;
+
+  /**
+   * The designated initializer.
+   */
+  public PathFinder() {
+    this.frontierComparator = new PathFinder.FrontierNodeComparator();
+  }
+
+  /**
+   * Finds the shortest path form point A to point B.
+   *
+   * @discussion Based on the A-star algorithm.
+   * 
+   * @param from  The node to start from.
+   * @param goal  The target node.
+   * @param graph The graph on which to find a path on.
+   *
+   * @return The shortest path from start to target.
+   */
+  public ArrayList<PathFindingNode> shortestPath(PathFindingNode from, PathFindingNode goal, PathFindingGraph graph) {
+    RequirementChecker requirementChecker = new RequirementChecker() {
+      public Boolean nodeMeetsRequirements(PathFindingNode node) {
+        return node == goal;
```

Comment:
I think you want something like node.equals(goal) and then override equals() in whichever class implements PathFindingNode